prow.sh

```diff
@@ -107,7 +135,24 @@ configvar CSI_PROW_GO_VERSION_GINKGO "${CSI_PROW_GO_VERSION_BUILD}" "Go version
 # kind version to use. If the pre-installed version is different,
 # the desired version is downloaded from https://github.com/kubernetes-sigs/kind/releases/download/
 # (if available), otherwise it is built from source.
-configvar CSI_PROW_KIND_VERSION "v0.6.0" "kind"
+#
+# Each version comes with a matching set of images, one per supported
+# Kubernetes release. Those image tags may get replaced in the future
+# with something that is incompatible with kind release that we are
+# using, therefore reference images with the digest. We need at least
+# one image digest for each major.minor Kubernetes version
+# number. Listing more than one patch version is okay but not
+# necessary, because always the most recent one will be used.
+#
+# Digests are part of the release notes at: https://github.com/kubernetes-sigs/kind/releases
+configvar CSI_PROW_KIND_VERSION "v0.7.0" "kind"
+configvar CSI_PROW_KIND_IMAGE_DIGEST_1_17_0 "sha256:9512edae126da271b66b990b6fff768fbb7cd786c7d39e86bdf55906352fdf62" "kind image digest for Kubernetes 1.17.0"
```

Comment:
Can we dynamically query this information? I want to start moving towards a goal where we  no longer need to update the testing framework for every new k8s release.

Reply:
> Can we dynamically query this information?

It comes from https://github.com/kubernetes-sigs/kind/releases, which isn't that easily parsed. The format might also change between releases.

Reply:
Instead of embedding config into our infrastructure and needing to update and sync it across all repos every release, can we do something instead like require the caller to pass in the kind version + k8s version (including patch)?

Reply:
That then makes it harder to run tests locally because the repo is no longer self-contained. At least I test that way from time to time.

There's also the potential problem that updating to, say, kind 0.8.0 will require changes that go beyond increasing the version number and the image digests. We had that with the config handling in the kind 0.5.0 -> 0.6.0 transition. We would have to move the entire `start_cluster` and `install_kind` functions into `test-infra`.

Reply:
I hope over time kind API will stabilize and we won't have this problem as frequently.

For ease of local testing, what if we just have one default? And if another version wants to be used, then the person needs to go to the kind releases to figure out the correct combinations. Or it falls back to building a node image.

I think it would also be good to ask kind folks if there's a way to make prebuild images more easily discoverable